insights/parsers/jboss_version.py

```diff
@@ -1,45 +1,60 @@
-================================================
-
-This module provides plugins access to file ``$JBOSS_HOME/version.txt``
-
-Typical content of file ``$JBOSS_HOME/version.txt`` is::
-
-    Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA
-
-This module parses the file content and stores data in the dict ``self.parsed``.
-The version info can also be got via ``obj.major`` and ``obj.minor``, etc.
-
-Examples:
-    >>> jboss_version.file_path
-    '/home/test/jboss/jboss-eap-6.4/version.txt'
-    >>> jboss_version.raw
-    'Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA'
-    >>> jboss_version.major
-    6
-    >>> jboss_version.minor
-    4
-    >>> jboss_version.release
-    3
-    >>> jboss_version.version
-    '6.4.3'
-    >>> jboss_version.code_name
-    'GA'
+JBoss version
+=============
+Provide information about the versions of all running Jboss on a system.
 """
 from .. import Parser, parser
 from ..specs import Specs
+import json
+from insights.tests import context_wrap
 
 
 @parser(Specs.jboss_version)
 class JbossVersion(Parser):
-    """Parses the content of file ``$JBOSS_HOME/version.txt``."""
+    """
+    JBoss version - File ``$JBOSS_HOME/version.txt``
+    ================================================
+
+    This class is to access to file ``$JBOSS_HOME/version.txt``
+
+    Typical content of file ``$JBOSS_HOME/version.txt`` is::
+
+        Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA
+
+    This class parses the file content and stores data in the dict ``self.parsed``.
+    The version info can also be got via ``obj.major`` and ``obj.minor``, etc.
+
+    Examples:
+        >>> jboss_version.file_path
+        '/home/test/jboss/jboss-eap-6.4/version.txt'
+        >>> jboss_version.raw
+        'Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA'
+        >>> jboss_version.major
+        6
+        >>> jboss_version.minor
+        4
+        >>> jboss_version.release
+        3
+        >>> jboss_version.version
+        '6.4.3'
+        >>> jboss_version.code_name
+        'GA'
+    """
 
     def parse_content(self, content):
         self.raw = content[0]
         product, _, version_name = [v.strip() for v in content[0].partition("- Version")]
-        version, code_name = version_name.strip().rsplit(".", 1)
-        major, minor, release = version.split(".")
+        if " GA" in version_name:
+            # handle Red Hat JBoss Web Server - Version 5.6 GA
+            version = version_name.split(' GA')[0]
+            code_name = "GA"
+            updated_version = version + ".0"
+            major, minor, release = updated_version.split('.')[0:3]
+        else:
+            # add empty code name for Red Hat Data Grid - Version 7.3.0
+            version_name = version_name.strip() + "."
+            major, minor, release, code_name = version_name.split(".")[0:4]
+            version = '.'.join([major, minor, release])
         self.parsed = {
```

Comment:
Since you modified the body of the `parse_content`, could you please 
1. Rename the `self.parsed` to `self._parsed` as a private member (may need to add one more property `product`)
2. Or add a docstring for it to make it a full public attribute?

Reply:
Updated.